# 3. Tamaño de clase de Universidad 
- Dataset: tamaño_de_clase.csv

Imaginemos una universidad donde la oficina del Decano publica datos que indican que el tamaño promedio de las clases es de 25 estudiantes. Sin embargo, al hablar con los estudiantes, muchos de ellos afirman que el tamaño promedio de las clases que han experimentado es más cercano a 28 estudiantes.


Calcule en python y explique:
1. Calcule en python el promedio desde todos los puntos de vista posibles
2. ¿Quién está sesgado en su cálculo del tamaño promedio de la clase, el Decano o el Estudiante?
3. Proporcione una solucion al sesgo.


In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import openpyxl

Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.
Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.


In [2]:
df_universidad = pd.read_csv("/Users/lolonastri/Desktop/POO_UCEMA/parciales/primer_parcial/ejercicio3/tamaño_de_clase.csv")
df_universidad

Unnamed: 0  Class_ID  Class_Size  Number_of_Students_Surveyed
0           0         1          20                            1
1           1         2          22                            1
2           2         3          23                            1
3           3         4          24                            1
4           4         5          25                            1
5           5         6          26                           10
6           6         7          27                           20
7           7         8          28                           30
8           8         9          29                           40
9           9        10          30                           50

___

In [7]:
df_universidad.shape

(10, 4)

In [10]:
# Distintas manera de calcular el promedio de Class_Size

df_universidad["Class_Size"].mean()

25.4

In [11]:
df_universidad["Class_Size"].sum()/df_universidad.shape[0]

25.4

In [12]:
df_universidad["Class_Size"].sum()/len(df_universidad)

25.4

___

# 2. ¿Quién está sesgado en el calculo del tamaño promedio de Class_Size, el Decano o el Estudiante? por que?

#### El Decano, porque el promedio de Class_Size es de 25.4 mientras que los alumnos dicen que el mismo es de 28.

___

# 3. Proporcione una solucion al sesgo.

#### Yo siento que lo que quizás está ocurriendo acá es un sesgo de respuesta ya que quizás las respuestas de los estudiantes (en caso de haberse recopilado los datos de esa manera) no son honestas o precisas debido al no conocimiento de los hechos. Por lo tanto, una solución sería que el Decano se acerque a los estudiantes y les explique la situación para que ellos puedan responder de manera más precisa y honesta.

#### Otra cosa que pudo haber sucedido es sesgo de selección. Esta ocurre cuando la selección de la muestra no es representativa de la población objetivo, lo que lleva a resultados sesgados como siento que es el caso.